src/lib/components/AuthCheck.svelte

```diff
@@ -0,0 +1,24 @@
+<script lang="ts">
```

Comment:
This is a client-side redirect component. Primarily for pages that don't need server-side sessions but still want to make sure the user is logged in. However there is a use case for using it on pages that do have server-side session auth even though it seems redundant: If the user logs out the `$user` object will change, the reactive code will run, and they will be redirected to the login page.